contracts/JBTieredNFTRewardDelegate.sol

```diff
@@ -0,0 +1,396 @@
+
+  /** 
+    @notice
+    The tier number of the provided token ID. 
+
+    @dev
+    Tier's are 1 indexed from the `tiers` array, meaning the 0th element of the array is tier 1.
+
+    @param _tokenId The ID of the token to get the tier number of. 
+
+    @return The tier number of the specified token ID.
+  */
+  function tierNumberOfToken(uint256 _tokenId) public view override returns (uint256) {
+    // Keep a reference to the number of tiers.
+    uint256 _numTiers = _tiers.length;
+
+    // Keep a reference to the tier being iterated on.
+    JBNFTRewardTier memory _tier;
+
+    for (uint256 _i; _i < _numTiers; ) {
+      // Set the tier being iterated on.
+      _tier = _tiers[_i];
+
+      // If the token ID is less than the tier's cieling and greater than this tier's starting, and greater than the previous tier's ceiling, it's in this tier.
+      if (
+        _tier.idCeiling > _tokenId && _tier.idCeiling - _tier.initialAllowance <= _tokenId
+        // 1 indexed, meaning the 0th element of the array is tier 1.
+      ) return _i + 1;
+
+      unchecked {
+        ++_i;
+      }
+    }
+
+    return 0;
+  }
+
+  //*********************************************************************//
+  // -------------------------- constructor ---------------------------- //
+  //*********************************************************************//
+
+  /**
+    @param _projectId The ID of the project for which this NFT should be minted in response to payments made. 
+    @param _directory The directory of terminals and controllers for projects.
+    @param _name The name of the token.
+    @param _symbol The symbol that the token should be represented by.
+    @param _tokenUriResolver A contract responsible for resolving the token URI for each token ID.
+    @param _baseUri The token's base URI, to be used if a URI resolver is not provided. 
+    @param _contractUri A URI where contract metadata can be found. 
+    @param _expectedCaller The address that should be calling calling the data source.
+    @param _owner The address that should own this contract.
+    @param _contributionToken The token that contributions are expected to be in terms of.
+    @param __tiers The tiers according to which token distribution will be made. 
+  */
+  constructor(
+    uint256 _projectId,
+    IJBDirectory _directory,
+    string memory _name,
+    string memory _symbol,
+    IToken721UriResolver _tokenUriResolver,
+    string memory _baseUri,
+    string memory _contractUri,
+    address _expectedCaller,
+    address _owner,
+    address _contributionToken,
+    JBNFTRewardTier[] memory __tiers
+  )
+    JBNFTRewardDataSource(
+      _projectId,
+      _directory,
+      _name,
+      _symbol,
+      _tokenUriResolver,
+      _baseUri,
+      _contractUri,
+      _expectedCaller,
+      _owner
+    )
+  {
+    contributionToken = _contributionToken;
+
+    // Make sure the tiers were delivered in order.
+    if (_tiers.length != 0) {
```

Comment:
this is a *pretty gnarly* bug ':)

Reply:
lol. womp. late last night I was changing the underscores around. gnarly indeed.

Reply:
feexed (allegedly)

Reply:
> lol. womp. late last night I was changing the underscores around. gnarly indeed.

Wait, there is more after I think ':) Maybe we should trailing instead of double leading